src/main/java/org/sonarlint/intellij/analysis/RunningAnalysesTracker.kt

```diff
@@ -44,6 +45,22 @@ class RunningAnalysesTracker {
         return analysisStateById[analysisId]
     }
 
+    fun cancel(analysisId: UUID) {
+        analysisStateById[analysisId]?.let {
+            it.cancel()
+            finish(it)
+        }
+    }
+
+    fun cancelSimilarAnalysis(analysisState: AnalysisState, console: SonarLintConsole) {
+        for (analysis in analysisStateById.values) {
+            if (analysis.isRedundant(analysisState)) {
+                console.info("Cancelling analysis ${analysis.id}")
+                analysis.cancel()
```

Comment:
Shouldn't we cancel the older ones, looks like we are cancelling the new one here

Reply:
We are iterating over `analysisStateById`, which contains all the running analysis, and we are cancelling each of them that are considered redundant against the one passed in the parameters